frontend/packages/dev-console/src/components/pipelines/pipeline-overview/TriggerLastRunButton.tsx

```diff
@@ -1,26 +1,35 @@
 import * as React from 'react';
 import { connect } from 'react-redux';
+import { Button } from '@patternfly/react-core';
 import { impersonateStateToProps } from '@console/internal/reducers/ui';
 import { useAccessReview } from '@console/internal/components/utils';
-import { Button } from '@patternfly/react-core';
+import { AccessReviewResourceAttributes } from '@console/internal/module/k8s';
 import { rerunPipelineAndStay } from '../../../utils/pipeline-actions';
-import { PipelineModel } from '../../../models';
+import { PipelineRunModel } from '../../../models';
 import { usePipelineRunWithUserLabel } from '../../pipelineruns/triggered-by';
 import { getLatestRun, PipelineRun } from '../../../utils/pipeline-augment';
 
 type TriggerLastRunButtonProps = {
   pipelineRuns: PipelineRun[];
+  namespace: string;
   impersonate?;
 };
 
 const TriggerLastRunButton: React.FC<TriggerLastRunButtonProps> = ({
   pipelineRuns,
+  namespace,
   impersonate,
 }) => {
   const latestRun = usePipelineRunWithUserLabel(
```

Comment:
While debugging the solution... I noticed a bug I accidentally made, can correct at the same time here?

In [pipelineruns/triggered-by/hooks.ts](https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/components/pipelineruns/triggered-by/hooks.ts#L29):

```
// Today
  return mergeLabelsWithResource(labels, plr);
// Suggested Change
  return plr && mergeLabelsWithResource(labels, plr);
```
That way it doesn't merge a label onto nothing creating an invalid k8s object.